### DISCUSSION TITLE
[@types/node] Node crypto misses option "jwk" from KeyFormat type

### DISCUSSION THREAD

**Discussion by KantiKuijk** [DefinitelyTyped/DefinitelyTyped#63491] [Issues with a @types package] [answered]
In @types/node the crypto.d.ts defines a KeyFormat type as:
`type KeyFormat = 'pem' | 'der';`
When using [crypto.createPublicKey](https://nodejs.org/api/crypto.html#cryptocreatepublickeykey), there is the option 'jwk', which is now not being recognised as valid.
Consequently, [crypto.verify](https://nodejs.org/api/crypto.html#cryptoverifyalgorithm-data-key-signature-callback) suffers the same fate (and possibly others too).

To fix this issue, I believe the previously mentioned line of code should be changed to:
`type KeyFormat = 'pem' | 'der' | 'jwk';`

The accompanying KeyObject is just as unhappy when supplying a working jwk object and complains about the kty property.

However, I only started working with typescript and the crypto module 2 days ago, so it is very possible I'm all wrong.

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment, marked as answer]
fixing in #63492
